Name a dataclip

midigofrank opened the pull request:
## Description

This PR *[adds/changes/fixes]... (A description of your work goes here.)*

Closes #311 

## Validation steps

1. *(How can a reviewer validate your work?)*

## Additional notes for the reviewer

1. *(Is there anything else the reviewer should know or look out for?)*

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [ ] Code generation (copilot but not intellisense)
- [ ] Learning or fact checking
- [ ] Strategy / design
- [ ] Optimisation / refactoring
- [ ] Translation / spellchecking / doc gen
- [ ] Other
- [ ] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

## Pre-submission checklist

- [ ] I have performed a **self-review** of my code.
- [ ] I have implemented and tested all related **authorization policies**. (e.g., `:owner`, `:admin`, `:editor`, `:viewer`)
- [ ] I have updated the **changelog**.
- [ ] I have ticked a box in "AI usage" in this PR

midigofrank commented:
@doc-han how do you rate my gpt-assisted React skills?

taylordowns2000 reviewed (changes requested):
UI looks great!

1. please add an audit trail event for the creation and removal of a label, then i think we're good
2. please add a flash message when a label is added "Label created. Dataclip will be saved permanently."
3. please add a flash message when a label is removed "Label removed. Dataclip will be removed when past your retention policy limit."

taylordowns2000 reviewed (changes requested):
This is looking great, but it seems as though the search doesn't return a named dataclip when that named dataclip is the _next_ cron run. In this image, I'd expect "long" to match and it does. I'd also expect "here" to match, but it returns no results.

<img width="431" height="339" alt="image" src="https://github.com/user-attachments/assets/5a636810-4749-4afe-aaba-b67cf840a543" />

This isn't likely a big problem, as the next cron run will almost never be named/saved. But it's still strange enough to warrant some investigation. Please see if you can get it to return when searching for the name of the next cron run.

midigofrank commented:
@taylordowns2000 this is likely the culprit https://github.com/OpenFn/lightning/pull/3409/files#diff-b400713bb29056407bd66ecb6e7a8663bde8d8adbf8a5a8f68f6bf3ffb8366eaR879

midigofrank commented:
Good catch @taylordowns2000 , the error was being caused by trying to match against the next-cron dataclip that doesn't have a name.

I have fixed the bug and added a test

elias-ba reviewed (approved):
Yo dude! this is a lot of JS/TS lol I love this. I clicked tested a bit and look at the code also. Everything is looking good to me. Nice implementation btw 👏🏽 ❤️

taylordowns2000 reviewed (approved):
thanks frank!
